ipn/ipnlocal/local.go

```diff
@@ -4355,6 +4366,12 @@ func (b *LocalBackend) enterStateLockedOnEntry(newState ipn.State, unlock unlock
 	}
 	b.pauseOrResumeControlClientLocked()
 
+	if newState == ipn.Running {
+		b.stopOfflineAutoUpdate()
```

Comment:
so here b.mu is locked, and then you call a func that acquires the the mutex.

so thus it must always be in this direction and documented.  (it would be a deadlock if the autoupdate-guarded code also could call into something that then acquired b.mu)

Reply:
with one small rearrangement, the new field is now protected by `b.mu` :+1: